bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,246 @@
+          rm -rf /etc/ipsec.conf /etc/ipsec.d /etc/ipsec.secrets
+          touch /etc/openvswitch/ipsec.conf
+          touch /etc/openvswitch/ipsec.secrets
+          mkdir -p /etc/openvswitch/ipsec.d
+          ln -s /etc/openvswitch/ipsec.conf /etc/ipsec.conf
+          ln -s /etc/openvswitch/ipsec.d /etc/ipsec.d
+          ln -s /etc/openvswitch/ipsec.secrets /etc/ipsec.secrets
+
+          /usr/libexec/ipsec/addconn --config /etc/openvswitch/ipsec.conf --checkconfig
+          # Check kernel modules
+          /usr/libexec/ipsec/_stackmanager start
+          # Check nss database status
+          /usr/sbin/ipsec --checknss
+          # Check nflog setup
+          /usr/sbin/ipsec --checknflog
+          # Start the pluto IKE daemon and use the ipsec configuration file that
+          # persists across container restarts. If the container has been restarted,
+          # this will contain the previous configuration from ovs-monitor-ipsec
+          /usr/libexec/ipsec/pluto --leak-detective --config /etc/openvswitch/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+
+          # Environment variables are for workaround for https://mail.openvswitch.org/pipermail/ovs-dev/2020-October/375734.html
+          # We now start ovs-monitor-ipsec which will monitor for changes in the ovs
+          # tunnelling configuration (for example addition of a node) and configures
+          # libreswan appropriately.
+          OVS_LOGDIR=/var/log/openvswitch OVS_RUNDIR=/var/run/openvswitch OVS_PKGDATADIR=/usr/share/openvswitch /usr/share/openvswitch/scripts/ovs-ctl --ike-daemon=libreswan start-ovs-ipsec
+
+          while true; do
+            sleep 60
+          done
+        env:
+        - name: OVS_LOG_LEVEL
+          value: info
+        - name: K8S_NODE
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/run/openvswitch
+          name: host-var-run-ovs
+        - mountPath: /var/log/openvswitch/
+          name: host-var-log-ovs
+        - mountPath: /etc/openvswitch
+          name: etc-openvswitch
+        terminationMessagePolicy: FallbackToLogsOnError
+        readinessProbe:
+          exec:
+            command:
+            - /bin/bash
+            - -c
+            - |
+              #!/bin/bash
+              ovs-appctl -t ovs-monitor-ipsec ipsec/status
+          initialDelaySeconds: 120
+          periodSeconds: 30
+          timeoutSeconds: 40
+        terminationGracePeriodSeconds: 10
+      nodeSelector:
+        beta.kubernetes.io/os: "linux"
+      volumes:
+      - name: host-var-log-ovs
+        hostPath:
+          path: /var/log/openvswitch
+      - name: host-var-run-ovs
+        hostPath:
+          path: /var/run/openvswitch
+      - name: signer-ca
+        configMap:
+          name: signer-ca
+      - name: etc-openvswitch
+        hostPath:
+          path: /var/lib/openvswitch/etc
```

Comment:
Note that this isn't `/etc/openvswitch` on the host. Is that OK for this use case?

I did this because `ovn-nbdb` was trying to write to `/etc/openvswitch` and getting selinux issues. But these are ovs components, not ovn.

Reply:
I just need a persistent host directory somewhere. I think I can leave this.

Reply:
(I don't know the right answer, just calling your attention to this).

Reply:
It should be ok. The host OVS instance doesnt need the files that we put in this directory. ovn-vsctl sets the string in ovsdb, and then ovs-monitor-ipsec reads that string. As long as this is persistent, it could be anywhere. I think I will leave it.